### PR TITLE
Add case of disk extension on lvm backend

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:1974547
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=disk_extension_lvm --guestname=RHEL.8.4.0,Win2019 --driveformat=virtio_scsi,virtio_blk --clone=no
(1/4) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (651.12 s)
 (2/4) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (531.55 s)
 (3/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (666.54 s)
 (4/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (605.11 s)

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha @menli820,Please help to review thanks.

File: qemu/tests/disk_extension_lvm.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    """
+    QEMU 'disk images extension on lvm in io-error status' test
+
+    1) Create small size lvm on large vg size.
+    2) Create qcow2 image based on the lvm
+    3) Boot vm with the lvm as vm data disk
+    4) Execute large size io on the data disk.
+    5) The vm will step in pause status due to no enough disk space.
+    6) Start to periodic increase lvm disk size(128M) at first pause.
+    7) Increase disk size when vm step in pause pause and resume vm.
```
Comment (zhenyzha) [inline qemu/tests/disk_extension_lvm.py:26]: 7) Increase disk size when vm step in pause and resume vm.

File: qemu/tests/disk_extension_lvm.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    if os_type == 'windows':
+        img_size = params["disk_size"]
+        guest_cmd = utils_misc.set_winutils_letter(session, guest_cmd)
+        disk = _get_window_disk_index_by_serail(disk_serial)
+        utils_disk.update_windows_disk_attributes(session, disk)
+        logging.info("Formatting disk:%s", disk)
+        driver = utils_disk.configure_empty_disk(session, disk, img_size,
+                                                 os_type)[0]
+        output_path = driver + ":\\test.dat"
+
```
Comment (zhenyzha) [inline qemu/tests/disk_extension_lvm.py:93]: No blank lines required

File: qemu/tests/disk_extension_lvm.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    else:
+        output_path = get_linux_drive_path(session, disk_serial)
+
+    if not output_path:
+        test.fail("Can not get output file path in guest.")
+
+    logging.debug("Get output file path %s", output_path)
+
+    guest_cmd = guest_cmd % output_path
+    wait_timeout = int(params.get("wait_timeout", 300))
```
Comment (zhenyzha) [inline qemu/tests/disk_extension_lvm.py:103]: Maybe putting it in row 62 will make it more unified?

File: qemu/tests/disk_extension_lvm.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    session = vm.wait_for_login(timeout=timeout)
+    if os_type == 'windows' and driver_name:
+        session = utils_test.qemu.windrv_check_running_verifier(session,
+                                                                vm,
+                                                                test,
+                                                                driver_name,
+                                                                timeout)
```
Comment (zhenyzha) [inline qemu/tests/disk_extension_lvm.py:83]: It's better to leave blank lines

File: qemu/tests/cfg/disk_extension_lvm.cfg [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    post_command += "pvremove -f `cat ${loopdev}` &&"
+    post_command += "losetup -d `cat ${loopdev}` &&"
+    post_command += "lvs /dev/${vgname}/${lvname} --noheadings; losetup -l &&"
+    post_command += "rm -rf ${tmpdir}"
+
+    extend_lvm_command = "lvextend --autobackup n --size +%s /dev/${vgname}/${lvname} >/dev/null"
+    get_lvm_size_command = "lvs /dev/${vgname}/${lvname} --noheadings --units=G|awk '{print $4}'|tr -d 'G'"
+
+
+
```
Comment (zhenyzha) [inline qemu/tests/cfg/disk_extension_lvm.cfg:57]: No need to leave more than one line

**Comment by zhenyzha** [comment]
git am 2866.patch
Applying: Add case of disk extension on lvm backend
.git/rebase-apply/patch:68: new blank line at EOF.
+
warning: 1 line adds whitespace errors.

**Comment by qingwangrh** [comment]
Patch updated.

**Comment by zhenyzha** [comment]
aarch64:
# git am 2866.patch
Applying: Add case of disk extension on lvm backend
 (1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.aarch64.io-github-autotest-qemu.disk_extension_lvm.arm64-pci: PASS (487.50 s)
ppc64le:
 (1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.disk_extension_lvm:PASS (559.38 s)

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Comment by menli820** [comment]
LGTM

 (1/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (1064.72 s)
 (2/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (1231.53 s)
 (3/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (728.30 s)
 (4/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (640.62 s)


 (1/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (950.56 s)
 (2/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (1010.31 s)
 (3/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (744.80 s)
 (4/4) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.disk_extension_lvm: PASS (901.00 s)

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**